src/main/java/org/altbeacon/beacon/service/BeaconService.java

```diff
@@ -90,7 +91,8 @@
     private boolean mBackgroundFlag = false;
     private ExtraDataBeaconTracker mExtraDataBeaconTracker;
     private ExecutorService mExecutor;
-
+    private final DistinctPacketDetector mDistinctPacketDetector = new DistinctPacketDetector();
```

Comment:
This needs to be on the service because the service class is where an 
AsyncTask is kicked to do scan processing on another thread.  This processing needs to be done there and not on the main thread as it is it is non-trivial.   This also causes the accessors to be needed on CycledLeScanner